File: src/Billing/Mediation/Engine/Models/ne.cs
// Ported from legacy Models_Mediation/ne.cs (MediationModel.ne — a switch / network element).
// SCALAR fields kept verbatim; EF navigation properties (bridgedroutes/jobs/routes/enumcdrformat/
// telcobrightpartner) and the collection-initialising ctor REMOVED (EF tech, not used by mediation).
// Most fields are legacy file-decode config and become vestigial under the routesphere feed.
#nullable disable

namespace MediationModel
{
    using System;
    using System.Collections.Generic;

    public partial class ne
    {
        public int idSwitch { get; set; }
        public int idCustomer { get; set; }
        public int idcdrformat { get; set; }
        public int idMediationRule { get; set; }
        public string SwitchName { get; set; }
        public string CDRPrefix { get; set; }
        public string FileExtension { get; set; }
        public string Description { get; set; }
        public string SourceFileLocations { get; set; }
        public string BackupFileLocations { get; set; }
        public Nullable<int> LoadingStopFlag { get; set; }
        public Nullable<int> LoadingSpanCount { get; set; }
        public Nullable<int> TransactionSizeForCDRLoading { get; set; }
        public Nullable<int> DecodingSpanCount { get; set; }
        public Nullable<int> SkipAutoCreateJob { get; set; }
        public Nullable<int> SkipCdrListed { get; set; }
        public Nullable<int> SkipCdrReceived { get; set; }
        public Nullable<int> SkipCdrDecoded { get; set; }
        public Nullable<int> SkipCdrBackedup { get; set; }
        public Nullable<int> KeepDecodedCDR { get; set; }
        public Nullable<int> KeepReceivedCdrServer { get; set; }
        public Nullable<int> CcrCauseCodeField { get; set; }
        public Nullable<int> SwitchTimeZoneId { get; set; }
        public string CallConnectIndicator { get; set; }
        public int FieldNoForTimeSummary { get; set; }
        public string EnableSummaryGeneration { get; set; }
        public int ExistingSummaryCacheSpanHr { get; set; }
        public int BatchToDecodeRatio { get; set; }
        public int FilterDuplicateCdr { get; set; }
        public int UseIdCallAsBillId { get; set; }
        public int PrependLocationNumberToFileName { get; set; }
        public int AllowEmptyFile { get; set; }
        public string ipOrTdm { get; set; }
    }
}

File: java/src/main/java/com/telcobright/billing/mediation/engine/models/ne.java
// Ported from legacy Models_Mediation/ne.cs (MediationModel.ne — a switch / network element).
// SCALAR fields kept verbatim; EF navigation properties (bridgedroutes/jobs/routes/enumcdrformat/
// telcobrightpartner) and the collection-initialising ctor REMOVED (EF tech, not used by mediation).
// Most fields are legacy file-decode config and become vestigial under the routesphere feed.
package com.telcobright.billing.mediation.engine.models;

import java.time.LocalDateTime;

public class ne {
    public int idSwitch;
    public int idCustomer;
    public int idcdrformat;
    public int idMediationRule;
    public String SwitchName;
    public String CDRPrefix;
    public String FileExtension;
    public String Description;
    public String SourceFileLocations;
    public String BackupFileLocations;
    public Integer LoadingStopFlag;
    public Integer LoadingSpanCount;
    public Integer TransactionSizeForCDRLoading;
    public Integer DecodingSpanCount;
    public Integer SkipAutoCreateJob;
    public Integer SkipCdrListed;
    public Integer SkipCdrReceived;
    public Integer SkipCdrDecoded;
    public Integer SkipCdrBackedup;
    public Integer KeepDecodedCDR;
    public Integer KeepReceivedCdrServer;
    public Integer CcrCauseCodeField;
    public Integer SwitchTimeZoneId;
    public String CallConnectIndicator;
    public int FieldNoForTimeSummary;
    public String EnableSummaryGeneration;
    public int ExistingSummaryCacheSpanHr;
    public int BatchToDecodeRatio;
    public int FilterDuplicateCdr;
    public int UseIdCallAsBillId;
    public int PrependLocationNumberToFileName;
    public int AllowEmptyFile;
    public String ipOrTdm;
}
